Check/Extensions/Duplicate.php

```diff
@@ -0,0 +1,151 @@
+    return dt('Check for duplicate extensions in the site codebase.');
+  }
+
+  /**
+   * Implements \SiteAudit\Check\Abstract\getResultFail().
+   */
+  public function getResultFail() {}
+
+  /**
+   * Implements \SiteAudit\Check\Abstract\getResultInfo().
+   */
+  public function getResultInfo() {}
+
+  /**
+   * Implements \SiteAudit\Check\Abstract\getResultPass().
+   */
+  public function getResultPass() {
+    return dt('No duplicate extensions were detected.');
+  }
+
+  /**
+   * Implements \SiteAudit\Check\Abstract\getResultWarn().
+   */
+  public function getResultWarn() {
+    $ret_val = dt('The following duplicate extensions were found:');
+    if (drush_get_option('html')) {
+      $ret_val = '<p>' . $ret_val . '</p>';
+      $ret_val .= '<table class="table table-condensed">';
+      $ret_val .= '<thead><tr><th>' . dt('Name') . '</th><th>' . dt('Paths') . '</th></thead>';
+      $ret_val .= '<tbody>';
+      foreach ($this->registry['extensions_dupe'] as $name => $paths) {
+        $ret_val .= '<tr><td>' . $name . '</td>';
+        $ret_val .= '<td>' . implode('<br/>', $paths) . '</td></tr>';
+      }
+      $ret_val .= '</tbody>';
+      $ret_val .= '</table>';
+    }
+    else {
+      foreach ($this->registry['extensions_dupe'] as $name => $paths) {
+        $ret_val .= PHP_EOL;
+        if (!drush_get_option('json')) {
+          $ret_val .= str_repeat(' ', 6);
+        }
+        $ret_val .= $name . PHP_EOL;
+        $extension_list = '';
+        foreach ($paths as $path) {
+          $extension_list .= str_repeat(' ', 8) . $path . PHP_EOL;
+        }
+        $ret_val .= rtrim($extension_list);
+      }
+    }
+    return $ret_val;
+  }
+
+  /**
+   * Implements \SiteAudit\Check\Abstract\getAction().
+   */
+  public function getAction() {
+    if ($this->score != SiteAuditCheckAbstract::AUDIT_CHECK_SCORE_PASS) {
+      return dt('Prune your codebase to have only one copy of any given extension. If you are using an installation profile, work with the maintainer to update the relevant modules. If you remove an enabled module, you may have to rebuild the registry.');
+    }
+  }
+
+  /**
+   * Implements \SiteAudit\Check\Abstract\calculateScore().
+   */
+  public function calculateScore() {
+    $this->registry['extensions_dupe'] = array();
+    $drupal_root = drush_get_context('DRUSH_SELECTED_DRUPAL_ROOT');
+    $settings = \Drupal::service('settings');
+    $kernel = \Drupal::service('kernel');
+    $command = "find $drupal_root -xdev -type f -name '*.info.yml' -o -path './" . $settings->get('file_public_path', $kernel->getSitePath() . '/files') . "' -prune";
+    exec($command, $result);
+
+    foreach ($result as $path) {
+      $path_parts = explode('/', $path);
+      $name = substr(array_pop($path_parts), 0, -9);
+      // Safe duplicates.
+      if (in_array($name, array(
+        'drupal_system_listing_compatible_test',
+        'drupal_system_listing_incompatible_test',
+        'aaa_update_test',
```

Comment:
I added it to Drupal 8 check. It was not in Drupal 7. On revmoving it, I get `aaa_update_test` in duplicate extensions. 
```
drush @d8 ae --detail
Extensions: 63%
  Count: Count the number of enabled extensions (modules and themes) in a site.
    There are 43 extensions enabled.
  Development: Check for enabled development modules.
    No enabled development extensions were detected; no action required.
  Duplicates: Check for duplicate extensions in the site codebase.
    The following duplicate extensions were found:
      aaa_update_test
        core/modules/update/tests/modules/aaa_update_test/aaa_update_test.info.yml (8.0.0-beta10)
        core/modules/update/tests/aaa_update_test/aaa_update_test.info.yml (8.0.0-beta10)
```